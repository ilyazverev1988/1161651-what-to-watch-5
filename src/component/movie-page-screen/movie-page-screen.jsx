import React, {Fragment, PureComponent} from "react";
import PropTypes from "prop-types";
import propsForFilms from "../../mocks/prop-types-for-films";
import propsForReviews from "../../mocks/prop-types-for-reviws";
import TabsForMoviePageScreen from "../tabs-for-movie-page-screen/tabs-for-movie-page-screen";
import withTabs from "../../hocs/with-tabs/with-tabs";
import MoreLikeThisFilm from "../more-like-this-film/more-like-this-film";
import withActiveItem from "../../hocs/with-active-item/with-active-item";
import {Link} from "react-router-dom";
import {addFilmToFavorite, fetchCommetsFilm} from "../../store/api-action";
import {store} from "../../index";
import {connect} from "react-redux";
import {returnFilmForID} from "../../utils";
import Avatar from "../avatar/avatar";

const MoreLikeFilms = withActiveItem(MoreLikeThisFilm);
const TabsInMoviePage = withTabs(TabsForMoviePageScreen);

export class MoviePage extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const id = this.props.match.params.id;
    store.dispatch(fetchCommetsFilm(id));
  }

  componentDidUpdate(prevProps) {
    const id = this.props.match.params.id;
    if (id !== prevProps.match.params.id) {
      store.dispatch(fetchCommetsFilm(id));
    }
  }

  render() {
    const {films, reviews, authorizationStatus} = this.props;
    const id = this.props.match.params.id;
    const {nameFilm, genre, releaseYear, filmCover, poster} = returnFilmForID(id, films);
    const filmsByGenre = films.filter((film) => film.genre === genre).slice(0, 4);
    return (
      <Fragment>
        <section className="movie-card movie-card--full">
          <div className="movie-card__hero">
            <div className="movie-card__bg">
              <img src={filmCover} alt={nameFilm}/>
            </div>
            <h1 className="visually-hidden">WTW</h1>
            <header className="page-header movie-card__head">
              <div className="logo">
                <Link to={`/`} className="logo__link">
                  <span className="logo__letter logo__letter--1">W</span>
                  <span className="logo__letter logo__letter--2">T</span>
                  <span className="logo__letter logo__letter--3">W</span>
                </Link>
              </div>

              <Avatar/>
            </header>

            <div className="movie-card__wrap">
              <div className="movie-card__desc">
                <h2 className="movie-card__title">{nameFilm}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{genre}</span>
                  <span className="movie-card__year">{releaseYear}</span>
                </p>

                <div className="movie-card__buttons">

                  <Link to={`/player/${this.props.match.params.id}`} className="btn btn--play movie-card__button" type="button">
                    <svg viewBox="0 0 19 19" width="19" height="19">
                      <use xlinkHref="#play-s"/>
                    </svg>
                    <span>Play</span>
                  </Link>

                  <Link to={`/mylist`} onClick={()=>{
                    store.dispatch(addFilmToFavorite(id));
                  }} className="btn btn--list movie-card__button" type="button">
                    <svg viewBox="0 0 19 20" width="19" height="20">
                      <use xlinkHref="#add"/>
                    </svg>
                    <span>My list</span>
                  </Link>
                  {authorizationStatus === `AUTH` ? <Link to={`/films/${this.props.match.params.id}/review`} href="add-review.html" className="btn movie-card__button">Add review</Link> : ``}
                </div>
              </div>
            </div>
          </div>

          <div className="movie-card__wrap movie-card__translate-top">
            <div className="movie-card__info">
              <div className="movie-card__poster movie-card__poster--big">
                <img src={poster} alt={nameFilm} width="218"
                  height="327"/>
              </div>

              <TabsInMoviePage film={returnFilmForID(id, films)} reviews={reviews}/>

            </div>
          </div>
        </section>
        <div className="page-content">
          <section className="catalog catalog--like-this">
            <h2 className="catalog__title">More like this</h2>
            <MoreLikeFilms films={filmsByGenre} />
          </section>

          <footer className="page-footer">
            <div className="logo">
              <a href="main.html" className="logo__link logo__link--light">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <div className="copyright">
              <p>Â© 2019 What to watch Ltd.</p>
            </div>
          </footer>
        </div>
      </Fragment>
    );
  }
}

MoviePage.propTypes = {
  films: PropTypes.arrayOf(propsForFilms).isRequired,
  reviews: propsForReviews,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  }),
  authorizationStatus: PropTypes.string.isRequired
};

const mapStateToProps = ({DATA, USER}) => ({
  reviews: DATA.comments,
  authorizationStatus: USER.authorizationStatus
});

export default connect(mapStateToProps, null)(MoviePage);
