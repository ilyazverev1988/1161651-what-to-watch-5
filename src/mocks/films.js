export default [
  {
    preview: `img/aviator.jpg`,
    nameFilm: `Aviator`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Thrillers`,
    releaseYear: 2014,
    descriptionFilm: `Friendship, metamorphosis, and adventure.`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Everything we do is connected with zen: sex, bliss, moonlight, sorrow.`,
    producer: `Heu, orexis!`,
    duration: `1h 39m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/dardjeeling-limited.jpg`,
    nameFilm: `Dardjeeling limited`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2016,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/bohemian-rhapsody.jpg`,
    nameFilm: `Bohemian rhapsody`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    nameFilm: `Fantastic beasts the crimes of grindelwald`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2018,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 50m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/johnny-english.jpg`,
    nameFilm: `Johnny english`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2019,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 9.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 30m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/macbeth.jpg`,
    nameFilm: `Macbeth`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2020,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 5.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 33m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/no-country-for-old-men.jpg`,
    nameFilm: `No country for old men`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2017,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 44m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/war-of-the-worlds.jpg`,
    nameFilm: `War of the worlds`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 54m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
];
