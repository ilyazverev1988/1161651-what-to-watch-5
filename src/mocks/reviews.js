export default [
  {
    film: `Aviator`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere1.`,
        filmScore: `1`,
        nameUser: `Ubi est albus1?`,
        dateReview: `December 24, 2018`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere2.`,
        filmScore: `2`,
        nameUser: `Ubi est albus2?`,
        dateReview: `December 20, 2018`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere3.`,
        filmScore: `3`,
        nameUser: `Ubi est albus3?`,
        dateReview: `December 21, 2018`
      }
    ]
  },
  {
    film: `Dardjeeling limited`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `4`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 10, 2018`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 20, 2019`
      },
    ]
  },
  {
    film: `Bohemian rhapsody`,
    review: [
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `3`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 13, 2018`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `1`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 21, 2011`
      }
    ]
  },
  {
    film: `Fantastic beasts the crimes of grindelwald`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `3`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 10, 2010`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 10, 2010`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `4`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 14, 2015`
      }
    ]
  },
  {
    film: `Johnny english`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `2`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 12, 2010`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `3`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 3, 2010`
      }
    ]
  },
  {
    film: `Macbeth`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 5, 2010`
      }
    ]
  },
  {
    film: `No country for old men`,
    review: [
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 6, 2010`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 26, 2010`
      }
    ]
  },
  {
    film: `War of the worlds`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 13, 2015`
      }
    ]
  },
];
