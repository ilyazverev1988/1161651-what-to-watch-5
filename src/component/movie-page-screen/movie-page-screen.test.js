import React from "react";
import renderer from "react-test-renderer";
import {MoviePage} from "./movie-page-screen";
import {MemoryRouter} from "react-router-dom";
import films from "../../mocks/films";
import reviews from "../../mocks/reviews";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const match = {params: {id: `4`}};
const mockStore = configureMockStore();
const store = mockStore({
  USER: {
    userData: {
      id: 1,
      email: `3@mail.ru`,
      name: `3`,
      avatarURL: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
      errorAuthorization: ``
    },
    authorizationStatus: `AUTH`
  },
});

const storeForSecondTest = mockStore({
  USER: {
    userData: {},
    authorizationStatus: `NO_AUTH`
  },
});

describe(`Should MoviePage render correctly`, () => {
  it(`With Autorization`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <MoviePage
                films={films}
                reviews={reviews}
                match={match}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Without Autorization`, () => {
    const tree = renderer
      .create(
          <Provider store={storeForSecondTest}>
            <MemoryRouter>
              <MoviePage
                films={films}
                reviews={reviews}
                match={match}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
