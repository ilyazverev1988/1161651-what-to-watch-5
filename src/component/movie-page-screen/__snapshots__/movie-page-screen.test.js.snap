// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MoviePage render correctly With Autorization 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Aviator4"
          src="img/bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <a
              href="/mylist"
              onClick={[Function]}
            >
              <img
                alt="User avatar"
                height="63"
                width="63"
              />
            </a>
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Aviator4
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thrillers
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/4"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <a
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              to="/mylist"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </a>
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Aviator4"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item  movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item  "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item  "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8.9
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                3
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              Friendship, metamorphosis, and adventure.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Wes Andreson
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Bill Murray, Jude Law and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Aviator1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Aviator2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Aviator3
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Aviator4
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Should MoviePage render correctly Without Autorization 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Aviator4"
          src="img/bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Aviator4
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thrillers
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/4"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <a
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              to="/mylist"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </a>
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Aviator4"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item  movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item  "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item  "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8.9
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                3
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              Friendship, metamorphosis, and adventure.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Wes Andreson
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Bill Murray, Jude Law and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Aviator1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Aviator2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Aviator3
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Aviator4
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
